File: 2022/test/day_04_test.exs
defmodule Day04Test do
  use ExUnit.Case

  import TestHelper
  import Day04

  test "part1" do
    assert part1(input("day_04_example_1")) == 2
    assert part1(input("day_04")) == 431
  end

  # test "part2" do
  #   assert part2(input("day_04_example_1")) == 123
  #   assert part2(input("day_04")) == 456
  # end
end

File: 2024/day04/day04_test.go
package day04

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

func TestPart1(t *testing.T) {
	assert.Equal(t, 18, Part1("input_example_1"))
	assert.Equal(t, 2530, Part1("input"))
}

func TestPart2(t *testing.T) {
	assert.Equal(t, 9, Part2("input_example_1"))
	assert.Equal(t, 1921, Part2("input"))
}
